### PR TITLE
feat(issues): Add zoom to new issue details graph

### PR DESCRIPTION
**Pull request by scttcper**
adds useCallback wrappers to useZoom
persists zoom changes

fixes https://www.notion.so/sentry/I-wish-I-can-use-the-graph-to-select-a-custom-timeframe-10c8b10e4b5d80c09430d38fef13090c

### CODE REVIEW

File: static/app/components/charts/useChartZoom.tsx [hunk 1]
```diff
@@ -78,51 +62,19 @@ interface Props {
  * the props that would be passed to the `BaseChart` as zoomRenderProps.
  */
 export function useChartZoom({
```
Comment (scttcper) [inline static/app/components/charts/useChartZoom.tsx:64]: this hook is only used in one place currently, took it as an opportunity to slim it down. The old ChartZoom did a lot to support onRestore which is not used anywhere

File: static/app/components/charts/useChartZoom.tsx [hunk 2]
```diff
@@ -78,51 +62,19 @@ interface Props {
...
-  disabled,
-}: Omit<Props, 'children'> = {}) {
-  const [currentPeriod, setCurrentPeriod] = useState<Period | undefined>({
-    period: period!,
-    start: getDate(start),
-    end: getDate(end),
-  });
-  const [history, setHistory] = useState<Period[]>([]);
-
-  const [zooming, setZooming] = useState<(() => void) | null>(null);
```
Comment (scttcper) [inline static/app/components/charts/useChartZoom.tsx:106]: using state here throws warnings and they're not used to rerender only for storing internal state.

File: static/app/components/charts/useChartZoom.tsx [hunk 3]
```diff
@@ -1,75 +1,59 @@
...
-  onChartReady?: EChartChartReadyHandler;
-  onDataZoom?: EChartDataZoomHandler;
-  onFinished?: EChartFinishedHandler;
-  onRestore?: EChartRestoreHandler;
-  onZoom?: (period: Period) => void;
-  period?: string | null;
+  onZoom?: (period: DateTimeUpdate) => void;
   router?: InjectedRouter;
+  /**
+   * Use eihter `saveOnZoom` or `usePageDate` not both
```
Comment (gggritso) [inline static/app/components/charts/useChartZoom.tsx:46]:
```suggestion
   * Use either `saveOnZoom` or `usePageDate` not both
```

File: static/app/components/charts/useChartZoom.tsx [hunk 4]
```diff
@@ -1,75 +1,59 @@
...
+   * Will persist zoom state to page filters
+   */
   saveOnZoom?: boolean;
   showSlider?: boolean;
-  start?: DateString;
+  /**
+   * Use eihter `saveOnZoom` or `usePageDate` not both
```
Comment (gggritso) [inline static/app/components/charts/useChartZoom.tsx:52]:
```suggestion
   * Use either `saveOnZoom` or `usePageDate` not both
```

**Review by gggritso — APPROVED** [review]

**Pull request merged**